Content.Server/Humanoid/Components/RandomHumanoidComponent.cs

```diff
@@ -0,0 +1,14 @@
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Humanoid.Components;
+
+/// <summary>
+///     This is added to a marker entity in order to spawn a randomized
+///     humanoid ingame.
+/// </summary>
+[RegisterComponent]
+public sealed class RandomHumanoidComponent : Component
+{
+    [DataField("settings")] public string RandomSettingsId = default!;
```

Comment:
This is slightly confusingly named, I would suggest either you rename ```RandomHumanoidPrototype``` to ```RandomHumanoidSettingsPrototype```,
 or you rename this variable to ```RandomHumanoidPrototypeId``` and the datafield to ```prototype```